cajita/src/Cajita_FastFourierTransform.hpp

```diff
@@ -471,11 +481,22 @@ class HeffteFastFourierTransform
         heffte_params.use_pencils = params.getPencils();
         heffte_params.use_reorder = params.getReorder();
 
-        // Set FFT options from given parameters
-        // heFFTe correctly handles 2D or 3D FFTs within "fft3d"
-        _fft = std::make_shared<heffte::fft3d<heffte_backend_type>>(
-            inbox, outbox, layout.localGrid()->globalGrid().comm(),
-            heffte_params );
+        // heffte recommends a sycl queue to avoid errors with SYCL backend
+        bool queue = params.getQueue();
+        // TODO: just one call here, no if/else
+        if ( queue )
```

Comment:
This will need to be an ifdef

Reply:
Alternatively, you can pass in a null pntr when not dealing with SYCL:
```
_fft = std::make_shared<heffte::fft3d<heffte_backend_type>>(
              (std::is_same<heffte_backend_type, heffte::backend::onemkl>::value) ?  q : nullptr, 
              inbox, outbox, layout.localGrid()->globalGrid().comm(), heffte_params );
```

Reply:
On another note, why use `shared_ptr` which has more overhead than the simpler `unique_ptr`.